image_processing.py

```diff
@@ -0,0 +1,66 @@
+import math
+import time
+
+import numpy as np
+from pycocotools.coco import COCO
+from scipy import misc
+
+data_dir = 'dataset'
+data_type = 'train2014'
+ann_file = '{0}/annotations/instances_{1}.json'.format(data_dir, data_type)
+output_shape = (244, 244, 3)
+
+
+def load_images():
+    '''Preprocessing of the MS Coco dataset
+        Output:
+            - ndarray (n, 244, 244, 3): the input data (x_train), images
+            - list (n): the labels of the images (y_train)
+    '''
+    x_train = list()
```

Comment:
Maybe it's faster to instantiate x_train with np.zeros(...) as we know the final size already. This way we allocate all memory at once.

I would have written X_train (capital X) because it is a matrix.